price-estimator/src/main.rs

```diff
@@ -64,22 +70,45 @@ fn main() {
     let web3 = web3_provider(&http_factory, options.node_url.as_str(), options.timeout).unwrap();
     // The private key is not actually used but StableXContractImpl requires it.
     let private_key = PrivateKey::from_raw([1u8; 32]).unwrap();
-    let _contract = Arc::new(
+    let contract = Arc::new(
         StableXContractImpl::new(&web3, private_key, 0)
             .wait()
             .unwrap(),
     );
 
-    // TODO: create event based orderbook
-    // TODO: handle http requests
+    let orderbook = EventBasedOrderbook::new(contract, web3, options.orderbook_file);
+    orderbook.initialize().wait().unwrap();
+    let orderbook = Arc::new(Orderbook::new(orderbook));
 
     let mut runtime = runtime::Builder::new()
```

Comment:
*edit*: I see that we have specific runtime configuration here. Maybe we should just call an `async run`. I think `tokio` provides `spawn` methods for spawning fibers in an `async` context.